addons/udes_stock/models/stock_move_line.py

```diff
@@ -425,6 +425,24 @@ def _swap_tracked_items(
                     ], order="id"
                 )
 
+                # For the message thread to notify of the tracked product swapped out
+                old_mls = mls_need_changing | swapped_lot_on_assigned_mls
+                for picking, mls in old_mls.groupby("picking_id"): 
+                    picking_qty = quantity_swapping
+                    swap_out_title = "Tracked Products Swapped Out: <ul>"
```

Comment:
the `end_of_list` variable too.

It might be worth just defining them inline at the end and not bother making them variable (as they aren't ever changed):
```python
body = "Tracked Products Swapped Out: <ul>" + swap_out_text + "</ul>"
```

Reply:
Changed, used the inline suggestion :slightly_smiling_face:

Reply:
I'm happy, will let @kdwyer resolve if happy with inline